### DISCUSSION TITLE
How to make a route guard or passing the route meta with this plugin?

### DISCUSSION THREAD

**Discussion by pheng0077** [posva/unplugin-vue-router#58] [Questions and Help] [answered]
I can't find the way to use route guards or passing route meta here.

**Comment by posva** [comment, marked as answer]
You can use `definePage()` within script setup:

```vue
<script setup>
definePage({
  meta: {}
})
</script>
```

It's still experimental because it has some limitations like not being able to use any other property within script setup

**Reply by pheng0077** [reply]
Hello, I'm using Vue3. So definePage function is unavailable. Are there any solutions for Vue3?

**Reply by posva** [reply]
This package exposes definePage

**Reply by pheng0077** [reply]
I've tried like this
`<script setup lang="ts">
definePage({
  meta: {
    auth: true
  }
})
</script>`

then I got this error 
`Uncaught ReferenceError: _mergeRouteRecord is not defined
    at routes:38:15
(anonymous) @ routes:38`

here is my router
`import { createRouter, createWebHistory } from "vue-router/auto";`
`const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
});`
`export default router;`

and vite version `3.0.4`

So did I do something wrong here?

**Reply by posva** [reply]
This likely got fixed a new recent release. You can check the playground for an example 🙂
